docs/source/about/contract.rst

```diff
@@ -0,0 +1,117 @@
+
+Basic requirements
+==================
+
+Codebase
+--------
+
+PaaSTA assumes there is one git repo and one resulting Docker image of that service. This image is deployed to multiple locations. This Docker image also may be used for slightly different use cases (worker daemon, web task, etc).
+
+Dependencies
+------------
+
+The Docker image MUST contain all the code necessary to run the service.
+
+State
+-----
+
+PaaSTA services SHOULD be stateless. Services MAY do filesystem IO, but all disks are ephemeral with Docker. (with the possible exception of RW bind-mounts, which are discouraged)
+
+Logs
+----
+
+Services should log to external log processors, at Yelp this is Scribe.
+
+Bouncing
+--------
+
+PaaSTA reserves the right to cause a bounce on your service at any time. Please make sure your service can handle this.
+See the docs on `bouncing <../workflow.html#bouncing>`_ for bounce settings that can help make bounces less impactful to your service.
+
+
+HTTP/TCP services
+-----------------
+
+* MUST be discoverable by SmartStack
+* MUST bind to port 8888 if it listens for traffic (e.g. if it runs a daemon that handles HTTP/TCP requests)
+
+Long-running tasks (services that don’t listen on a port, or “batch daemons”)
+-----------------------------------------------------------------------------
+
+MUST EITHER:
+
+* Not have a SmartStack configuration file (smartstack.yaml)...
+* If they have a SmartStack configuration file (e.g. because a single service codebase provides both an HTTP service and a long-running task) the instance configuration for the long-running task MUST NOT define a proxy_port::
+
+    # marathon.yaml
+    ---
+    main:
+      instances: 3
+    worker:
+      instances: 1
+    # smartstack.yaml
+    main:
+      proxy_port: 12345
+    # (no worker definition in smartstack.yaml!)
+
+* MAY set ``healthcheck_mode`` to ``cmd`` and  specify a ``healthcheck_cmd`` in `marathon-<cluster>.yaml <yelpsoa_configs.html#marathon-clustername-yaml>`_ to give Mesos better insight into the health of a task::
+
+    # marathon.yaml
+    ---
+    gearman_worker:
+      healthcheck_mode = "cmd"
+      healthcheck_cmd = "/some_status_command.py"
+
+Deployment Workflow
+===================
+
+Jenkins pipeline (Recommended)
+------------------------------
+
+Most services will want to use Jenkins to drive the deploy process. You can think of our Jenkins tooling as a reference implementation of the manual steps (below). Services that want this:
+
+* MUST contain a Makefile. This is how Jenkins will manipulate the service.
+* The Makefile SHOULD delegate non-trivial work to helper scripts written in an appropriate language
+   * Helper scripts MAY live in paasta/ and MAY be named after the Makefile targets that call them, e.g. paasta/test, paasta/itest,  etc.
+* MUST respond to make test
+* SHOULD respond by running unit tests
```

Comment:
It's not clear that this point is referring to the instruction above. To be more explicit:

SHOULD respond to ``make test`` by running unit tests.